### PR TITLE
nixos/k3s: Add environmentFile as an option for configuring

### PR DESCRIPTION
**Pull request by asonix**
###### Description of changes
Add environmentFile as an option for configuring k3s. This makes it easier to add configurations with secrets, e.g. managing a postgres connection string containing a password with `sops`

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [x] ~~(Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes~~ not needed anymore?
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/doc/manual/release-notes/rl-2305.section.md [hunk 1]
```diff
@@ -256,3 +256,5 @@ In addition to numerous new and upgraded packages, this release has the followin
 - The option `services.nomad.extraSettingsPlugins` has been fixed to allow more than one plugin in the path.
 
 - The option `services.prometheus.exporters.pihole.interval` does not exist anymore and has been removed.
+
+- `k3s` can now be configured with an EnvironmentFile for it's systemd service, allowing secrets to be provided without ending up in the Nix Store.
```
Comment (yajo) [inline nixos/doc/manual/release-notes/rl-2305.section.md:260]: Little typo

```suggestion
- `k3s` can now be configured with an EnvironmentFile for its systemd service, allowing secrets to be provided without ending up in the Nix Store.
```

**Review by yajo — CHANGES_REQUESTED** [review]
The patch looks good except for a small typo (see below).

However, the options already include `tokenFile`. That's been the only secret I needed to deploy a k3s cluster. What's the use case you have in mind? I mean... what more secrets would you need?

**Comment by asonix** [comment]
Specifically for postgres connection strings such as `postgres://username:password@host:port/database`. k3s supports multiple data stores for it's state and configuring those from the extra options would leak those secrets into the nix store

**Comment by asonix** [comment]
And thanks, I can never remember when to use `its` vs `it's`

**Review by yajo — APPROVED** [review]
OK that makes total sense.

**Review by superherointj — APPROVED** [review]

**Pull request merged**